pkg/opentofu/tofutoschema.go

```diff
@@ -114,25 +132,67 @@ func hydrateObjectSchema(name string, ty cty.Type, sch *schema.Schema) {
 	slices.Sort(sch.Required)
 }
 
-func hydrateMapSchema(name string, ty cty.Type, sch *schema.Schema) {
-	sch.Title = name
+func hydrateMapSchema(sch *schema.Schema, name string, ty cty.Type, defaults *typeexpr.Defaults) {
 	sch.Type = "object"
 	sch.PropertyNames = &schema.Schema{
 		Pattern: "^.*$",
 	}
 	sch.AdditionalProperties = new(schema.Schema)
-	hydrateSchemaFromNameAndType("", ty.ElementType(), sch.AdditionalProperties.(*schema.Schema))
+	hydrateSchemaFromNameTypeAndDefaults(sch.AdditionalProperties.(*schema.Schema), "", ty.ElementType(), getDefaultChildren(name, defaults))
 }
 
-func hydrateArraySchema(name string, ty cty.Type, sch *schema.Schema) {
-	sch.Title = name
+func hydrateArraySchema(sch *schema.Schema, name string, ty cty.Type, defaults *typeexpr.Defaults) {
 	sch.Type = "array"
 	sch.Items = new(schema.Schema)
-	hydrateSchemaFromNameAndType("", ty.ElementType(), sch.Items)
+	hydrateSchemaFromNameTypeAndDefaults(sch.Items, "", ty.ElementType(), getDefaultChildren(name, defaults))
 }
 
-func hydrateSetSchema(name string, ty cty.Type, sch *schema.Schema) {
-	hydrateArraySchema(name, ty, sch)
+func hydrateSetSchema(sch *schema.Schema, name string, ty cty.Type, defaults *typeexpr.Defaults) {
+	hydrateArraySchema(sch, name, ty, defaults)
 	sch.UniqueItems = true
-	hydrateSchemaFromNameAndType("", ty.ElementType(), sch.Items)
+	hydrateSchemaFromNameTypeAndDefaults(sch.Items, "", ty.ElementType(), getDefaultChildren(name, defaults))
+}
+
+func ctyValueToInterface(val cty.Value) interface{} {
+	valJSON, err := ctyjson.Marshal(val, val.Type())
+	if err != nil {
+		// Should never happen, since all possible known
+		// values have a JSON mapping.
+		panic(fmt.Errorf("failed to serialize default value as JSON: %s", err))
+	}
+	var def interface{}
+	err = json.Unmarshal(valJSON, &def)
+	if err != nil {
+		// Again should never happen, because valJSON is
+		// guaranteed valid by ctyjson.Marshal.
+		panic(fmt.Errorf("failed to re-parse default value from JSON: %s", err))
+	}
+	removeNullKeys(def)
+	return def
+}
+
+func getDefaultChildren(name string, defaults *typeexpr.Defaults) *typeexpr.Defaults {
+	var children *typeexpr.Defaults
+	if defaults != nil {
+		if attDefaultVal, exists := defaults.Children[name]; exists {
+			children = attDefaultVal
+		}
+	}
+	return children
+}
+
+func removeNullKeys(defVal interface{}) {
```

Comment:
Add a brief comment about what this function does and why it exists. It's not immediately clear.